trinity/sync/common/events.py

```diff
@@ -0,0 +1,26 @@
+from typing import (
+    Type,
+    Union,
+)
+
+from lahja import (
+    BaseEvent,
+    BaseRequestResponseEvent,
+)
+
+from trinity.sync.common.types import (
+    SyncProgress
+)
+
+
+class SyncingResponse(BaseEvent):
+
+    def __init__(self, syncing: Union[bool, SyncProgress]) -> None:
+        self.syncing = syncing
```

Comment:
What if this was split into two properties.

```python
class SyncingResponse(BaseEvent):
    is_syncing: bool
    progress: Optional[SyncProgress]
```

This should give slightly better type safety in the places where it is used so that you don't have to do a type check on `self.syncing`.